lang/java/reef-tests/src/main/java/org/apache/reef/tests/fail/task/Driver.java

```diff
@@ -135,6 +135,13 @@ public void onNext(final RunningTask task) {
             + " not equal expected ID " + taskId);
       }
 
+      // In case of slow machines, it takes some time until task is ready.
+      try {
+        Thread.sleep(1000);
```

Comment:
This makes me very nervous. Whenever we have to add real time `sleep`s to the code, we are basically hiding a bigger synchronization problem without fixing it. Similar code to the one in this test might well be in applications, so I'd rather fix the underlying issue than add a bandaid to the test.

When we raise a `RunningTask`, we better make sure the Task is ready for whatever method calls we provide on the `RunningTask` instance.

What does it mean for the task not to be ready? Is the network connection not established yet? Or is it not running yet?